### PR TITLE
Config return

### PR DESCRIPTION
**Pull request by znicholls**
Add ability to include config in output

- [x] Tests added
- [x] Documentation added
- [x] Example added (in the documentation, to an existing notebook, or in a new notebook) (N/A)
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openscm/openscm-runner/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,7 +22,8 @@
     "fair>=1.6.1a0,<2.0.0",
     "matplotlib==3.2.2",
     "pyam-iamc",
-    "pymagicc>=2.0.0rc6",
+    # "pymagicc>=2.0.0rc7,<3",
+    "pymagicc @ git+https://github.com/openscm/pymagicc.git@fix-naming",
```
Comment (lewisjared) [inline setup.py:26]: Should we roll back this change if openscm/pymagicc#311 is being postponed?

File: src/openscm_runner/adapters/magicc7/magicc7.py [hunk 2]
```diff
@@ -16,8 +16,13 @@
 LOGGER = logging.getLogger(__name__)
 
 
-# TODO: upgrade pymagicc and remove this
-VARIABLE_MAP = {"Ocean Heat Uptake": "HEATUPTK_AGGREG"}
```
Comment (lewisjared) [inline src/openscm_runner/adapters/magicc7/magicc7.py:20]: This would also need to be added back

File: src/openscm_runner/run.py [hunk 3]
```diff
@@ -33,9 +50,9 @@ def run(
     output_variables : list[str]
         Variables to include in the output
 
-    full_config : bool
-        Include the configuration used to run each model in the output's
-        metadata
+    out_config : dict[str: tuple of str]
+        Dictionary where each key is a model and each value is a list of
```
Comment (lewisjared) [inline src/openscm_runner/run.py:54]: The typing states a tuple of str instead of a list

File: src/openscm_runner/run.py [hunk 4]
```diff
@@ -11,11 +11,28 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def _check_out_config(out_config, climate_models_cfgs):
+    if out_config is not None:
+        unknown_models = set(out_config.keys()) - set(climate_models_cfgs.keys())
+        if unknown_models:
```
Comment (lewisjared) [inline src/openscm_runner/run.py:17]: Why not silently ignore? Say we had a script to run MAGICC and Fair, but for testing purposes, we wanted to just run Fair. We would have to keep the out_config and climate_models_cfgs in lockstep. Not a huge problem and very much a style preference, but I think L83 is sufficient.

**Review by lewisjared — CHANGES_REQUESTED** [review]
Probably should back out the fix-naming changes unless they are needed the `openscm-runner`

File: setup.py (hunk 1 above)
Comment (znicholls) [reply setup.py:26]: Sorry I think we should push openscm/pymagicc#311 through. The one change which isn't in openscm/pymagicc#311  is renaming 'Surface Temperature' to 'Surface Air Tempreature Change'. I keep flip-flopping about whether to include that or not.

File: src/openscm_runner/run.py (hunk 3 above)
Comment (znicholls) [reply src/openscm_runner/run.py:54]:
```suggestion
        Dictionary where each key is a model and each value is a tuple of
```

File: src/openscm_runner/run.py (hunk 4 above)
Comment (znicholls) [reply src/openscm_runner/run.py:17]: Yep good point. I'll add some extra logging around line 83 so debugging is easier (i.e. to make it not completely silent) but I agree this is probably overkill so will remove.

Comment (znicholls) [reply src/openscm_runner/run.py:17]: I think my logic here was that if you have a key in `out_config` which isn't in `climate_models_cfgs`, something is obviously wrong. I hadn't thought about the case where you only want to run one so this is a good point.

File: setup.py (hunk 1 above)
Comment (lewisjared) [reply setup.py:26]: Hmmm that is a rather large change... I couldn't count the number of times I have typed `Surface Temperature` across our code base

File: src/openscm_runner/run.py (hunk 4 above)
Comment (znicholls) [reply src/openscm_runner/run.py:17]: Just pushed updates, see what you think

File: setup.py (hunk 1 above)
Comment (znicholls) [reply setup.py:26]: Yep, I was thinking we do openscm/pymagicc#311 without Surface Temperature. Then release v2. Then add a deprecation warning that Surface Temperature will change to Surface Air Temperature Change in v3. Then go through that pain next year.

Comment (lewisjared) [reply setup.py:26]: That agrees with my comment on openscm/pymagicc#311 so happy to proceed.  I agree we shouldn't make the switch right now.

**Review by lewisjared — APPROVED** [review]
LGTM waiting on pymagicc@v2.0.0rc7

**Pull request merged**